android/src/main/java/io/mosip/tuvali/transfer/Chunker.kt

```diff
@@ -61,7 +62,8 @@ class Chunker(private val data: ByteArray, private val maxDataBytes: Int) :
   }
 
   fun isComplete(): Boolean {
-    val isComplete = chunksReadCounter > (totalChunkCount - 1).toInt()
+    Log.i(logTag,"chunksReadCounter: $chunksReadCounter")
+    val isComplete = chunksReadCounter > (totalChunkCount - 1)
```

Comment:
Taking the above example, it should be greater than 4  because chunksReadCounter is a postIncrement counter which is managed by next(). So once you read the 5th chunk (i.e. the last chunk), the value of chunksReadCounter will become 5 and not 4. So now when we call the isComplete(), we should get true. And above logic complies to that. 

But, I agree that the logic is hard to understand and we can simplify it to chunksReadIndex == totalChunkCount.